src/hacbs/components/Commits/CommitSidePanel.tsx

```diff
@@ -0,0 +1,170 @@
+import React from 'react';
+import {
+  DrawerActions,
+  DrawerHead,
+  DrawerPanelContent,
+  DrawerCloseButton,
+  DrawerPanelBody,
+  DropdownItem,
+  Dropdown,
+  DropdownToggle,
+  Text,
+  Button,
+  ButtonVariant,
+} from '@patternfly/react-core';
+import AngleDownIcon from '@patternfly/react-icons/dist/js/icons/angle-down-icon';
+import AngleUpIcon from '@patternfly/react-icons/dist/js/icons/angle-up-icon';
+import { runStatus } from '../../../shared';
+import { Timestamp } from '../../../shared/components/timestamp/Timestamp';
+import { PipelineRunKind } from '../../types';
+import { showPLRMessage, showPLRType, statuses } from '../../utils/commits-utils';
+
+export interface SortedPLRList {
+  [runStatus.Running]: PipelineRunKind[];
+  [runStatus.Cancelled]: PipelineRunKind[];
+  [runStatus.Failed]: PipelineRunKind[];
+  [runStatus.Pending]: PipelineRunKind[];
+  [runStatus.Succeeded]: PipelineRunKind[];
+}
+
+interface CommitSidePanelProps {
+  drawerRef: any;
+  isExpanded: boolean;
+  onPanelCloseClick: () => void;
+  initialStatus: string;
+  selectedPipelineRun: PipelineRunKind;
+  sortedPLRList: SortedPLRList;
+  setSelectedPipelineRun: (plr: PipelineRunKind) => void;
+}
+
+const CommitSidePanel: React.FC<CommitSidePanelProps> = ({
+  drawerRef,
+  isExpanded,
+  onPanelCloseClick,
+  initialStatus,
+  setSelectedPipelineRun,
+  sortedPLRList,
+}) => {
+  const [currentStatus, setCurrentStatus] = React.useState<string>(initialStatus);
```

Comment:
In this case, If you are expecting to sync your props (initialStatus) with useState variable then this will fail to sync. UseState will not re-initialise when the props change. You need to sync it either with useEffect or re-render this component with `key`.

Reply:
Also the fact that this status will change on progression of events, better to call something other than `initialStatus`, maybe `currentStatus` ?

Reply:
Renamed to commitStatus and moved inside useEffect